src/main/java/org/springframework/integration/kafka/inbound/KafkaMessageDrivenChannelAdapter.java

```diff
@@ -152,28 +195,43 @@ public void setFilterInRetry(boolean filterInRetry) {
 	protected void onInit() {
 		super.onInit();
 
-		AcknowledgingMessageListener<K, V> listener = this.listener;
+		if (this.mode.equals(ListenerMode.record)) {
+			AcknowledgingMessageListener<K, V> listener = this.recordListener;
 
-		boolean filterInRetry = this.filterInRetry && this.retryTemplate != null && this.recordFilterStrategy != null;
+			boolean filterInRetry = this.filterInRetry && this.retryTemplate != null && this.recordFilterStrategy != null;
 
-		if (filterInRetry) {
-			listener = new FilteringAcknowledgingMessageListenerAdapter<>(listener, this.recordFilterStrategy,
-					this.ackDiscarded);
-			listener = new RetryingAcknowledgingMessageListenerAdapter<>(listener, this.retryTemplate,
-					this.recoveryCallback);
+			if (filterInRetry) {
+				if (this.recordFilterStrategy != null) {
```

Comment:
This condition is redundant, because `filterInRetry` can be true only when `this.recordFilterStrategy != null` as well.